cmd/hive/clustersync_test.go

```diff
@@ -0,0 +1,396 @@
+				} else {
+					*v = csList // Assuming csList is populated normally
+				}
+			}
+			return returnErr
+		}).Times(callTimes) // Expect List to be called n times based on the error condition
+}
+
+func TestPrintFailingCluster(t *testing.T) {
+	tests := []struct {
+		name        string
+		cdList      *hivev1.ClusterDeploymentList
+		csList      *v1alpha1.ClusterSyncList
+		expectError bool
+	}{
+		{
+			name: "Successful_Execution",
+			cdList: &hivev1.ClusterDeploymentList{
+				Items: []hivev1.ClusterDeployment{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster",
+						Namespace: "uhc-production-test",
+					},
+				}},
+			},
+			csList: &v1alpha1.ClusterSyncList{
+				Items: []v1alpha1.ClusterSync{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster-sync",
+						Namespace: "uhc-production-test",
+					},
+				}},
+			},
+			expectError: false,
+		},
+		{
+			name: "Missing_ClusterDeployment",
+			cdList: &hivev1.ClusterDeploymentList{
+				Items: []hivev1.ClusterDeployment{},
+			},
+			csList: &v1alpha1.ClusterSyncList{
+				Items: []v1alpha1.ClusterSync{},
+			},
+			expectError: true,
+		},
+		{
+			name: "SyncSet_Failure",
+			cdList: &hivev1.ClusterDeploymentList{
+				Items: []hivev1.ClusterDeployment{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster",
+						Namespace: "uhc-production-test",
+					},
+				}},
+			},
+			csList: &v1alpha1.ClusterSyncList{
+				Items: []v1alpha1.ClusterSync{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster-sync",
+						Namespace: "uhc-production-test",
+					},
+					Status: v1alpha1.ClusterSyncStatus{
+						SyncSets: []v1alpha1.SyncStatus{{
+							Name:           "sync-failure",
+							Result:         "Failure",
+							FailureMessage: "Some error occurred",
+						}},
+					},
+				}},
+			},
+			expectError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			scheme := runtime.NewScheme()
+			if err := hivev1.AddToScheme(scheme); err != nil {
+				t.Fatalf("Failed to add ClusterDeployment to scheme: %v", err)
+			}
+
+			if err := v1alpha1.AddToScheme(scheme); err != nil {
+				t.Fatalf("Failed to add ClusterSync to scheme: %v", err)
+			}
+
+			// Add ClusterDeployments and ClusterSyncs to the fake client
+			objects := []client.Object{}
+			for _, cd := range tt.cdList.Items {
+				objects = append(objects, &cd)
+			}
+			for _, cs := range tt.csList.Items {
+				objects = append(objects, &cs)
+			}
+
+			client := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(objects...).
+				Build()
+
+			o := &clusterSyncFailuresOptions{
+				kubeCli:   client,
+				clusterID: "test",
+			}
+
+			err := o.printFailingCluster()
+			if tt.expectError {
+				assert.Error(t, err)
+				return
+			}
+
+			assert.NoError(t, err)
+
+		})
+	}
+}
+
+func testListFailingClusterSyncs(t *testing.T, returnErr error, expectedResults []failingClusterSync, isEmpty bool) {
```

Comment:
I still feel this should be inlined in `TestListFailingClusterSyncs`, but if you want to keep it as a helper function could you rename it to something else?  Someone could easily think that this should be a unit test with a lower-case `t` type and then uppercase it.   Having it the same case-insensitive name as the actual unit test is a bit confusing.

Reply:
Renamed the helper function to avoid confusion.